tachyon/math/base/semigroups.h

```diff
@@ -216,6 +324,107 @@ class AdditiveSemigroup {
     }
     return ret;
   }
+
+  // scalar: s
+  // bases: [G₀, G₁, ..., Gₙ₋₁]
+  // return: [sG₀, sG₁, ..., sGₙ₋₁]
+  template <typename F,
+            typename ReturnTy =
+                typename internal::AdditiveSemigroupTraits<G>::ReturnTy>
+  static std::vector<ReturnTy> ScalingBases(const F& scalar,
```

Comment:
How about "ConstantScaleBases"?

Reply:
How about "ScaleBasesWithScalar"? MulScalarAndBasesInBatch?

Reply:
When thinking about `BatchInverse()`, `ScalarMul()`, `MSM(MultiScalarMultiplication)`, i think it's good to `BatchScalarMul()` and let compiler differentiate the cases (I discussed with @dongchangYoo )